### PR TITLE
mimirtool: update livecheck

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by p-linnane** [comment]
Not the latest release yet: https://github.com/grafana/mimir/releases

File: Formula/m/mimirtool.rb [hunk 1]
```diff
@@ -7,9 +7,13 @@ class Mimirtool < Formula
   license "AGPL-3.0-only"
   head "https://github.com/grafana/mimir.git", branch: "main"
 
+  # Upstream appears to use GitHub releases to indicate that a version is
+  # released (and some tagged versions don't end up as a release), so it's
+  # necessary to check release versions instead of tags.
   livecheck do
     url :stable
     regex(/^mimir-(\d+\.\d+\.\d+)$/i)
```
Comment (samford) [inline Formula/m/mimirtool.rb:15]:
```suggestion
    regex(/^mimir[._-]v?(\d+(?:\.\d+)+)$/i)
```

Upstream reliably uses three-part versions but these changes bring the regex in line with our typical standards.

**Review by samford — COMMENTED** [review]
Hmm, the "latest" GitHub release is currently 2.10.4 but there's a subsequent [2.10.5 release](https://github.com/grafana/mimir/releases/tag/mimir-2.10.5) (which the formula uses) that wasn't marked as "latest". At the moment, the `GithubLatest` strategy will return 2.10.4, so the formula version will appear to be newer than the livecheck version. The trouble is, 2.10.5 doesn't appear in the `GithubReleases` results for whatever reason, so it also returns 2.10.4 as the newest release.

**Comment by chenrui333** [comment]
> Hmm, the "latest" GitHub release is currently 2.10.4 but there's a subsequent [2.10.5 release](https://github.com/grafana/mimir/releases/tag/mimir-2.10.5) (which the formula uses) that wasn't marked as "latest"

I filed an issue for it, I think it is more of semantics issue.

**Review by p-linnane — APPROVED** [review]

**Pull request merged**